pkg/daemon/controller/controller.go

```diff
@@ -315,6 +329,10 @@ func (sc *ServiceController) expectedServiceCommand(expected servicescm.Service)
 // resolveNodeVariables returns a map, with the keys being each variable, and the value being the string to replace the
 // variable with
 func (sc *ServiceController) resolveNodeVariables(svc servicescm.Service) (map[string]string, error) {
+	if sc.nodeName == "" {
+		return nil, errors.New("instance has no associated node")
+	}
```

Comment:
I feel this is unnecessary as this will be caught below in the Get(), and this is not going to cause a panic as it is not a pointer. I don't think its a good pattern to validate that struct fields (excluding pointers), are set before they are used in a method.

This should never occur, instead the service CM validation should ensure that bootstrap services don't rely on node variables, which is what you are doing in the other commit